app/lib/fake/backend/fake_pub_worker.dart

```diff
@@ -0,0 +1,254 @@
+  // doing whatever work they planned to do.
+  await instancesDeleted;
+
+  // Stop the task backend, and instance execution
+  await Future.wait([
+    taskBackend.stop(),
+    cloud.stopInstanceExecution(),
+  ]);
+}
+
+/// Updates the task status for all packages and imitates
+/// pub_worker using fake pana and dartdoc results.
+Future<void> processTasksWithFakePanaAndDartdoc() async {
+  final fakePanaRunner = FakePanaRunner();
+  await fork(() async {
+    final fakeCloudCompute = FakeCloudCompute(
+      instanceRunner: (instance) async {
+        final payload = Payload.fromJson(
+          json.decode(instance.arguments.first) as Map<String, dynamic>,
+        );
+        for (final v in payload.versions) {
+          final client = httpClientWithAuthorization(
+            tokenProvider: () async => v.token,
+            sessionIdProvider: () async => null,
+            csrfTokenProvider: () async => null,
+          );
+          try {
+            final api = PubApiClient(payload.pubHostedUrl, client: client);
+            await withTempDirectory((tempDir) async {
+              final packageStatus = await scoreCardBackend.getPackageStatus(
+                  payload.package, v.version);
+
+              final dartdocFiles = fakeDartdocFiles(payload.package, v.version);
+              final docData = PubDartdocData.fromJson(json.decode(
+                dartdocFiles['pub-data.json']!,
+              ) as Map<String, dynamic>);
+              final docSection = documentationCoverageSection(
+                documented: docData.coverage?.documented ?? 0,
+                total: docData.coverage?.total ?? 0,
+              );
+
+              late Summary summary;
+              if (packageStatus.isObsolete ||
+                  packageStatus.isLegacy ||
+                  packageStatus.isDiscontinued) {
+                summary = _emptySummary(payload.package, v.version);
+                dartdocFiles.clear();
+              } else {
+                final s = await fakePanaRunner.analyze(
+                  package: payload.package,
+                  version: v.version,
+                  packageStatus: packageStatus,
+                );
+                final updatedReport = s.report?.joinSection(docSection);
+                summary = s.change(report: updatedReport);
+              }
+
+              final r = await api.taskUploadResult(payload.package, v.version);
+
+              final blobFile = File(p.join(tempDir.path, 'files.blob'));
+              final builder = IndexedBlobBuilder(blobFile.openWrite());
+
+              Future<void> addFileAsStringGzipped(
+                  String path, String content) async {
+                final stream =
+                    Stream.fromIterable([gzip.encode(utf8.encode(content))]);
+                await builder.addFile(path, stream);
+              }
+
+              for (final e in dartdocFiles.entries) {
+                await addFileAsStringGzipped(e.key, e.value);
```

Comment:
The `.html` files created by `pub_worker` are actually JSON files :rofl: 

This happened when we moved the postprocessing/sanitizing into the sandbox

Reply:
We can probably just hardcode some JSON that fits the DartDocPage pattern

Reply:
Updated. PTAL.